src/jit/compiler.hpp

```diff
@@ -1456,20 +1456,38 @@ inline void GenTree::SetOper(genTreeOps oper, ValueNumberUpdate vnUpdate)
     }
 }
 
-inline void GenTree::CopyFrom(const GenTree* src, Compiler* comp)
+//------------------------------------------------------------------------
+// CopyFrom: replace this with the src node. If the source is isolated node, then use it as it.
+//           Otherwise, make a src clone.
+//
+// Arguments:
+//    src - source tree, that replaces this.
+//    comp - the compiler instanse to transfer annotations for arrays.
+//
+inline void GenTree::CopyFrom(GenTree* src, Compiler* comp)
 {
     /* The source may be big only if the target is also a big node */
 
     assert((gtDebugFlags & GTF_DEBUG_NODE_LARGE) || GenTree::s_gtNodeSizes[src->gtOper] == TREE_NODE_SZ_SMALL);
-    GenTreePtr prev = gtPrev;
-    GenTreePtr next = gtNext;
```

Comment:
I would suggest retaining this, and clearing the next/prev on 'this' when/if you clone it. Then you avoid the situation where there's some remaining link to 'src' in the tree, with those fields set to those old values.

Reply:
Ok, I will create a fix for such places in two separate PRs, because there are many changes there.
This PR will wait until I merge fixes.

Reply:
There is also DEBUG_DESTROY_NODE which is the recommended way to mark a node that should no longer be referenced in the graph of nodes.